spec-draft.md

```diff
@@ -136,6 +136,106 @@ Example payload for setting "Airplane Mode":
+(height), and Z is extensive (depth). We arbitrarily stipulate the "default"
+position for a device (i.e., its position when the X, Y and Z angles are 0, 0,
+and 0) is as follows:
+
+* The device screen is facing us, parallel to the value 0 on the Z axis (i.e.,
+  it extends into the positive Z space, not the negative (see below))
+* The bottom edge of the device lies on the X axis The left edge of the device
+  lies on the Y axis
+
+In terms of polarity, we stipulate that the X axis goes from negative to
+positive, left to right. The Y axis goes from negative to positive, bottom to
+top. And the Z axis goes from negative to positive, front to back.
+
+Now, assuming the bottom, left, front edge of the device always remains at the
+origin point (i.e., the point x=0, y=0, z=0), we can define any rotation. The
+"default" rotation described above would be equivalent to a normal "portrait"
+orientation, the device being held straight out in front of the user. The
+rotation `(x=90deg, y=0deg, z=0deg)` signifies a similar portrait orientation,
+but "face up" on a table, for example. "Face down portrait" would then be
+`(x=90deg, y=0deg, z=180deg)`.
+
+We can now define the necessary endpoints:
+
+* GET /session/:sessionid/rotation
+    * returns DeviceRotation
+* POST /session/:sessionid/rotation
+    * accepts a DeviceRotation
+
+The remote end MUST reply with the capability "deviceRotationEnabled" in order
+to use these methods.
+
+### DeviceRotation
+
+`DeviceRotation` is an object with 3 keys, `x`, `y`, and `z`. Each of these
+represents an angle in degrees (on the real number scale `0 <= deg < 360`).
+This is how rotation works for each axis:
+
+* X axis: angles increasing towards the positive Z space (i.e., 'away')
+* Y axis: angles increasing towards the positive Z space (i.e., 'left')
+* Z axis: angles increasing towards the negative Y space (i.e., 'down')
+
+Example payloads for setting various common rotations:
+
+* Standard portrait
+    * `{"name": "rotation", "parameters": {"x": 0, "y": 0, "z": 0}}`
+* Portrait face-up
+    * `{"name": "rotation", "parameters": {"x": 90, "y": 0, "z": 0}}`
+* Portrait face-down
+    * `{"name": "rotation", "parameters": {"x": 90, "y": 0, "z": 180}}`
+    * `{"name": "rotation", "parameters": {"x": 270, "y": 0, "z": 0}}`
+* Portrait, slight away-tilt
+    * `{"name": "rotation", "parameters": {"x": 15, "y": 0, "z": 0}}`
+* Landscape right
+    * `{"name": "rotation", "parameters": {"x": 0, "y": 0, "z": 90}}`
+* Landscape left
+    * `{"name": "rotation", "parameters": {"x": 0, "y": 0, "z": 270}}`
+
+### Errors
+
+If the `DeviceRotation` object sent in by the client does not conform to the
+specification (e.g., if values are non-numeric or out of the allowed range), or
+if the server cannot perform the requested specific rotation (say because it is
+limited to a certain subset of rotations that does not include the requested
+orientation), it MUST respond with an `UnableToRotateDevice` error (code XXX).
```

Comment:
Code will be `Unable to Rotate Device`. See [WebDriver Errors](http://w3c.github.io/webdriver/webdriver-spec.html#handling-errors)